src/exporter.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Stanford University see LICENSE for license
+
+import honeybadger from 'honeybadger'
+import config from 'config'
+import fs from 'fs'
+import { query } from './sinopia_client'
+
+const Honeybadger = honeybadger.configure({
+  apiKey: process.env.HONEYBADGER_API_KEY
+})
+
+const getDateString = () => {
+  return (new Date()).toISOString()
+}
+
+const initAndGetSavePath = (groupName, exportBasePath) => {
+  let dateTag
+  if (!exportBasePath) {
+    exportBasePath = config.get('exportBasePath')
+    dateTag = getDateString()
+  }
+
+  let savePathString = [exportBasePath, groupName].join('/')
+  if (dateTag)
+    savePathString = [savePathString, dateTag].join('_');
+
+  fs.mkdirSync(savePathString, { recursive: true })
+  return savePathString
+}
+
+const reportError = (errorObject, consoleErrMessage = null) => {
+  Honeybadger.notify(errorObject, {
+    context: {
+      argv: process.argv,
+      sinopiaApiBasePath: config.get('sinopia_api.basePath')
+    }
+  })
+
+  if (consoleErrMessage)
+    console.error(consoleErrMessage)
+}
+
+export const exportGroup = async (groupName, containingDir = '') => {
+  console.info(`beginning export of RDF from group: ${groupName}`)
+
+  const savePathString = initAndGetSavePath(groupName, containingDir)
+  let done = false
+  let uri = [config.get('sinopia_api.basePath'), `/resource/?limit=${config.get('sinopia_api.query_limit')}&start=0&group=`, groupName].join('')
+
+  while (!done) {
+    let result_count = 0
+    await query(uri, { Accept: 'application/json' }).then((groupResources) => {
+      result_count = groupResources.data.length
+
+      groupResources.data.map((resource) => {
```

Comment:
I think this is going to parallelize here.

Reply:
Can you explain why that would be an issue? This already has the resources returned as is writing to disk.